### DISCUSSION TITLE
Unclear formulation in the help concerning ExtractByIndex

### DISCUSSION THREAD

**Discussion by tozGitHub** [WinEdt-Team/WinEdt-Support#30] [2.2 Issues, Bugs and Problems]
Dear Alex, dear community,

WinEdt Macro Manual|String Manipulation|ExtractByIndex
Maybe here my problem is that English is not my native language, but it was not clear to me whether End is inclusive or exclusive (does ExtractByIndex(0, "Source", 0, 3); extract three or four letters?)

For example the  Python code
```Python
"source"[0:3]
```
gives
```
'sou'
```
While WinEdt is different:
```
ExtractByIndex(0, "Source", 0, 3);Prompt("%!0");
```
yields
<img width="266" alt="Pasted image 20240218152907" src="https://github.com/WinEdt-Team/WinEdt-Support/assets/52357589/1af2d886-af2a-4a67-84df-f4312b805fcd">

WinEdt versions tested: Build: 20230519  (v. 11.1) - 64-bit and Build: 20240319  (v. 11.2) - 64-bit
Windows version: 10.0.22631 Build 22631.

Best,
Thomas

**Comment by Alex-Simonic** [comment]
Yes, the End index is a part of the extracted string. This has now been clarified:

Do extraction to this position including the End index (zero-based). The length of the extracted string is End-Start+1 or zero if End is less than Start.
